### ISSUE TITLE
35network-manager won't setup network if other modules don't add any "initqueue/finished" hook

### ISSUE THREAD

**Issue by ryncsn** [dracutdevs/dracut#694]
Previously when using 35network-legacy, it will add initqueue/finished hooks to wait for the interface to be configured. However 35network-manager won't do so, and if other modules didn't add any initqueue/finished hook, initqueue won't run, and the nm-run.sh won't have a chance to run either.

Is 35network-manager expecting other module to install initqueue/finished to wait for network to get ready, or perhaps 35network-manager should at lease ensure nm-run.sh got executed for at least once?

**Comment by ryncsn** [comment]
There is a workaround for this problem in kdump, by install another initqueue hook to keep the initqueue running until network is ready:

See: https://lists.fedorahosted.org/archives/list/kexec@lists.fedoraproject.org/thread/ZAYFCE3UOH7UTCBFWZ6KGR5VI625IGZT/

**Comment by cgwalters** [comment]
xref https://github.com/coreos/fedora-coreos-config/pull/259
and in particular https://mail.gnome.org/archives/networkmanager-list/2020-January/msg00000.html
where I argue that we should be running NM as a regular systemd unit.

**Comment by haraldh** [comment]
@lkundrak !

**Comment by LorbusChris** [comment]
Any updates on this?

**Comment by jomeier** [comment]
@cgwalters 
Hi. Any news on this one? Greets, Josef

**Comment by jomeier** [comment]
@haraldh 
Hi Harald. @LorbusChris mentioned that you could come to the rescue here.

Could you give priority to this issue, please? The OKD community needs you 🕺🏻

**Comment by LorbusChris** [comment]
I think I meant @lkundrak might be able to do that.
Anyway, now we've pinged 'em all! :tongue: 
And as @jomeier said, we over at Fedora CoreOS and OKD would appreciate a fix for this issue =)

**Comment by LorbusChris** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1809641 is a related BZ for this

**Comment by tyll** [comment]
Hi, The NetworkManager team is happy to help. What is the exact request?

@thom311 or @bengal please take a look.

> https://bugzilla.redhat.com/show_bug.cgi?id=1809641 is a related BZ for this

This is not reported against NetworkManager and also does not include the NM team in CC. Should it?

**Comment by LorbusChris** [comment]
Thanks @tyll! The root cause for that BZ is this issue here :)
We need this to be solved to unblock https://github.com/coreos/fedora-coreos-config/pull/259 (and subsequently a few other things)

**Comment by bengal** [comment]
@LorbusChris I'm looking into this.

**Comment by jlebon** [comment]
To be clear, https://bugzilla.redhat.com/show_bug.cgi?id=1809641 isn't an issue with NM in the initrd, it's with the current legacy network dracut module FCOS is still using. We have a hunch that switching to NM will resolve this too, but it's not sure.

IOW, we should focus on getting the original problem filed here fixed so we can unblock FCOS moving to NM in the initrd (https://github.com/coreos/fedora-coreos-config/pull/259). And that *may* fix that RHBZ at the same time. If not, that's a separate issue to fix.
